### ISSUE TITLE
1.8.2 broke Docker build for linux_amd64 target

### ISSUE THREAD

**Issue by manno** [goreleaser/goreleaser#3057]
### What happened?

The docker build failed to work, until we switch the target from "linux_amd64" to "linux_amd64_v1".

```
   ⨯ release failed after 5.22s error=failed to build ghcr.io/epinio/epinio-server:v0.7.0-amd64: exit status 1: #1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 328B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/alpine:latest
#3 DONE 1.1s

#4 [certs 1/2] FROM docker.io/library/alpine@sha256:4edbd2beb5f78b1014028f4fbb99f3237d9561100b6881aabbf5acce2c4f9454
#4 DONE 0.0s

#5 [internal] load build context
#5 transferring context: 2B done
#5 DONE 0.0s

#6 [certs 2/2] RUN apk --update --no-cache add ca-certificates
#6 CACHED

#7 [stage-1 1/2] COPY --from=certs /etc/ssl/certs /etc/ssl/certs
#7 CACHED

#8 [stage-1 2/2] COPY epinio /
#8 ERROR: "/epinio" not found: not found
------
 > [stage-1 2/2] COPY epinio /:
------
error: failed to solve: failed to compute cache key: "/epinio" not found: not found

```

The same happens for "linux_amd64_v2".

### How can we reproduce this?

Our configuration can be found here: https://github.com/epinio/epinio/blob/main/.goreleaser.yml

A simpler config that is failing looks like this:

```
---
project_name: epinio

archives:
  - format: binary

before:
  hooks:
    - go mod download

builds:
  - id: epinio
    main: ./main.go
    binary: epinio
    ldflags:
      - -w -s
      - -X "github.com/epinio/epinio/internal/version.Version={{ .Tag }}"
    goos:
      - linux
    goarch:
      - amd64
    goarm:
      - "7"
    targets:
    - linux_amd64
    #- linux_amd64_v2

changelog:
  skip: true

env:
  - CGO_ENABLED=0

snapshot:
  name_template: "{{ .Tag }}-next"

dockers:
  -
    use: buildx
    goos: linux
    goarch: amd64
    ids:
    - epinio
    image_templates:
    - "ghcr.io/epinio/epinio-server:{{ .Tag }}-amd64"
    dockerfile: images/Dockerfile
    build_flag_templates:
    - "--pull"
    - "--label=org.opencontainers.image.created={{.Date}}"
    - "--label=org.opencontainers.image.title={{.ProjectName}}"
    - "--label=org.opencontainers.image.revision={{.FullCommit}}"
    - "--label=org.opencontainers.image.version={{.Version}}"
    - "--label=org.opencontainers.image.source=https://github.com/epinio/epinio"
    - "--build-arg=DIST_BINARY=epinio"
    - "--platform=linux/amd64"
```

### goreleaser version

```bash
Tested locally with


goreleaser version 1.8.2
commit: a53afde02ad8d4d088125a6bbbe23e11399ded0e
built by: homebrew
goos: darwin
goarch: arm64

https://goreleaser.com
```

Also happens when using the Github action with "latest".

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

This seems to fix it for us: https://github.com/epinio/epinio/pull/1368

**Comment by caarlos0** [comment]
Will be fixed in https://github.com/goreleaser/goreleaser/pull/3056

that said, you can avoid these issues by using goos, goarch, etc instead of targets.

**Comment by manno** [comment]
> Will be fixed in #3056
> 
> that said, you can avoid these issues by using goos, goarch, etc instead of targets.

Ah, I only added targets, because it seemed more straight forward than the ignore list: https://github.com/epinio/epinio/commit/e89d164fd907c222d14d22cf09b4f2c74720c8f1
Maybe that wasn't a good idea.
